### ISSUE TITLE
get_symbol_info omits readonly/ref modifiers on structs (IsReadOnly/IsRefLikeType ignored)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#747]
## Summary

[`get_symbol_info`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/src/RoslynMcp.Core/Query/GetSymbolInfoOperation.cs) builds `DetailedSymbolInfo.Modifiers` via private `GetModifiers`. For **named types** it already surfaces `record` when `INamedTypeSymbol.IsRecord` is true, and it surfaces field/method `readonly` via `IFieldSymbol.IsReadOnly` / `IMethodSymbol.IsReadOnly` — but it **never** checks `INamedTypeSymbol.IsReadOnly` or `INamedTypeSymbol.IsRefLikeType`.

So `readonly struct`, `ref struct`, and `readonly ref struct` (and `readonly record struct`) are reported with the same modifiers as a plain mutable struct (plus `record` when applicable). Clients cannot tell stack-only / immutable-by-default value types apart from ordinary structs via `modifiers`.

Contract docs on [`DetailedSymbolInfo.Modifiers`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/src/RoslynMcp.Contracts/Models/DetailedSymbolInfo.cs) describe modifiers as language modifiers (`static`, `abstract`, `sealed`, `virtual`, `override`, **etc.**). The implementation already treats `record` as such an etc. for types; `readonly` / `ref` on structs are the same class of keyword and are already handled for fields/methods — only the named-type arms are missing.

## Bug

In [`GetSymbolInfoOperation.GetModifiers`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/src/RoslynMcp.Core/Query/GetSymbolInfoOperation.cs) (~L131–159):

```csharp
if (symbol is IMethodSymbol method)
{
    if (method.IsAsync) modifiers.Add("async");
    if (method.IsReadOnly) modifiers.Add("readonly");  // method-level readonly present
}

if (symbol is IFieldSymbol field)
{
    if (field.IsConst) modifiers.Add("const");
    if (field.IsReadOnly) modifiers.Add("readonly");   // field-level readonly present
    if (field.IsVolatile) modifiers.Add("volatile");
}

if (symbol is INamedTypeSymbol namedType)
{
    if (namedType.IsRecord) modifiers.Add("record");
    // missing: namedType.IsReadOnly → "readonly"
    // missing: namedType.IsRefLikeType → "ref"
}
```

Top-level `ISymbol` flags (`IsStatic`, `IsAbstract`, `IsSealed`, …) do **not** cover struct `readonly` / `ref`; those live on `INamedTypeSymbol` only.

## Expected

- `readonly struct Point { }` → `modifiers` contains **`readonly`** (and not `ref`).
- `ref struct SpanLike { }` → `modifiers` contains **`ref`** (and not `readonly` unless also readonly).
- `readonly ref struct SpanLike { }` → `modifiers` contains **`readonly`** and **`ref`**.
- `readonly record struct Point(int X);` → `modifiers` contains **`record`** and **`readonly`** (Kind may still be `Struct` until `#746` lands — orthogonal).
- Plain `struct` / `record struct` without those keywords keep today's modifiers (no spurious `readonly`/`ref`).

## Actual

On master `dddf25bee962a80158cb7fda8ceebb5314ca9930` (Release v0.5.2):

- `get_symbol_info` on `readonly struct Point { }` → `modifiers` **omits** `readonly` (typically `[]` or only accessibility-related flags already covered elsewhere).
- `get_symbol_info` on `ref struct SpanLike { }` → `modifiers` **omits** `ref`.
- `get_symbol_info` on `readonly record struct Point(int X);` → `modifiers` includes **`record`** but still **omits** `readonly`.
- Same operation already adds `readonly` for `readonly` fields and `readonly` members inside structs — type-level gap only.

## Evidence

- Implementation: [`src/RoslynMcp.Core/Query/GetSymbolInfoOperation.cs`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/src/RoslynMcp.Core/Query/GetSymbolInfoOperation.cs) `GetModifiers` ~L131–159 (named-type arm only checks `IsRecord`).
- Contract: [`DetailedSymbolInfo.Modifiers`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/src/RoslynMcp.Contracts/Models/DetailedSymbolInfo.cs) — “Modifiers (static, abstract, sealed, virtual, override, etc.).”
- Sibling asymmetry: field/method `IsReadOnly` → `"readonly"` already; type `IsRecord` → `"record"` already; type `IsReadOnly` / `IsRefLikeType` ignored.
- Tests: [`GetSymbolInfoToolTests`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/dddf25bee962a80158cb7fda8ceebb5314ca9930/tests/RoslynMcp.Server.Tests/Tools/GetSymbolInfoToolTests.cs) are schema/workspace smoke only — no regression for struct `readonly`/`ref` modifiers.
- Master SHA: `dddf25bee962a80158cb7fda8ceebb5314ca9930` (Release v0.5.2).
- Dup check: open outline `#648` / `#649` are **syntax Kind** on `get_document_outline` (want ReadonlyStruct / RefStruct Kind strings), not semantic `modifiers` on `get_symbol_info`. Open `#746` is Kind `Struct` vs `Record` for record structs via `SymbolKindMapper` — orthogonal to modifier strings. Open `get_symbol_info` tickets `#435`/`#536`/`#621` are identifier/reserved-keyword validation only. No prior ticket that `GetModifiers` skips type-level `readonly`/`ref`.

## Repro sketch

1. Solution with:
   ```csharp
   public readonly struct Point { public int X; }
   public ref struct SpanLike { public int X; }
   public readonly record struct Rec(int X);
   ```
2. `get_symbol_info` on each type name.
3. Inspect `modifiers`: expect `readonly` / `ref` / (`record`+`readonly`); actual omits type-level `readonly`/`ref`.

## Relation

- Incorrect-behavior modifiers gap on the **query** path `get_symbol_info`, not outline Kind.
- Orthogonal to outline `#648`/`#649` (syntax Kind discrimination), `#746` (semantic Kind Record vs Struct for record structs), and rename/move Kind `#743`/`#744`/`#745`.
- No open leftover (label **bug** only; open leftover count 0). Leave Dependabot alone.

## Separate follow-ups

- `IPropertySymbol.IsReadOnly` / `IsRequired` (and field `IsRequired`) are also absent from `GetModifiers` — file separately if desired; do not fold into this minimum named-type fix unless deliberately batching.
- Whether outline `#648` should mirror these as Kind strings vs leaving discrimination to `get_symbol_info.modifiers` — product/schema for outline; out of scope here.
- Do **not** fold `#746`, outline `#648`/`#649`, or validation `#435`/`#536`/`#621` into this minimum fix.

## Acceptance criteria

- [ ] `get_symbol_info` on `readonly struct` includes `"readonly"` in `modifiers`.
- [ ] `get_symbol_info` on `ref struct` includes `"ref"` in `modifiers`.
- [ ] `get_symbol_info` on `readonly ref struct` includes both `"readonly"` and `"ref"`.
- [ ] `readonly record struct` still includes `"record"` and also `"readonly"`.
- [ ] Plain struct / non-readonly record struct do not gain spurious `readonly`/`ref`.
- [ ] Regression test(s) covering the cases above.
- [ ] Do not treat outline `#648`/`#649`, Kind `#746`, or property/`required` modifier gaps as in-scope for the minimum fix (unless explicitly batching).

## Suggested fix direction (high-level)

In the `INamedTypeSymbol` arm of `GetModifiers`:

```csharp
if (namedType.IsReadOnly) modifiers.Add("readonly");
if (namedType.IsRefLikeType) modifiers.Add("ref");
if (namedType.IsRecord) modifiers.Add("record");
```

(Order can match C# declaration order `readonly`/`ref` before `record`, or keep `record` first — pick one and test it.) Keep property/`required` and outline Kind work as separate follow-ups.
